sky/serve/autoscalers.py

```diff
@@ -16,32 +17,37 @@
 
 logger = sky_logging.init_logger(__name__)
 
-_UPSCALE_DELAY_S = 300
-_DOWNSCALE_DELAY_S = 1200
-
 
 class AutoscalerDecisionOperator(enum.Enum):
     SCALE_UP = 'scale_up'
     SCALE_DOWN = 'scale_down'
-    NO_OP = 'no_op'
 
 
 @dataclasses.dataclass
 class AutoscalerDecision:
     """Autoscaling decisions.
 
-    |---------------------------------------------------------|
-    | Operator   | TargetType | Meaning                       |
-    |------------|------------|-------------------------------|
-    | SCALE_UP   | int        | Number of replicas to add     |
-    |------------|------------|-------------------------------|
-    | SCALE_DOWN | List[int]  | List of replica ids to remove |
-    |------------|------------|-------------------------------|
-    | NO_OP      | None       | No scaling needed             |
-    |---------------------------------------------------------|
+    |------------------------------------------------------------------------|
+    | Operator   | TargetType                | Meaning                       |
+    |------------|---------------------------|-------------------------------|
+    | SCALE_UP   | Optional[Dict[str, Any]   | Resource override to add      |
+    |------------|---------------------------|-------------------------------|
+    | SCALE_DOWN | int                       | Replica id to remove          |
+    |------------------------------------------------------------------------|
     """
     operator: AutoscalerDecisionOperator
-    target: Optional[Union[int, List[int]]]
+    target: Union[Optional[Dict[str, Any]], int]
+
+    # TODO(MaoZiming): Add a doc to elaborate on autoscaling policies.
+    def __init__(self,
+                 operator: AutoscalerDecisionOperator,
+                 target: Union[Optional[Dict[str, Any]], int] = None):
+
+        # Scale down requires replica ids to remove.
+        assert not (operator == AutoscalerDecisionOperator.SCALE_DOWN and
+                    target is None)
```

Comment:
Remove it or check it more strictly? i.e.

```python
assert (op == SCALE_UP and (tar is None or isinstance(tar, dict))) or (op == SCALE_DOWN and isinstance(tar, int))
```

Reply:
If we decided to remove it, we could delete the whole constructor since `dataclasses` already have a default ctor.

Reply:
let's check it more strictly. Thanks!